### ISSUE TITLE
krun/kast/ktest triggers guice creation error

### ISSUE THREAD

**Issue by yilongli** [kframework/k-legacy#776]
This seems like a serious problem. I just typed `krun` and got the following:

```
Exception in thread "main" com.google.inject.CreationException: Guice creation errors:

1) Error in custom provider, org.kframework.utils.errorsystem.KExceptionManager$KEMException
  at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
  at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
  while locating org.kframework.kil.loader.Context
  at org.kframework.utils.inject.DefinitionLoadingModule.kompileOptions(DefinitionLoadingModule.java:48)
  while locating org.kframework.kompile.KompileOptions
  at org.kframework.krun.KRunModule$CommonModule.getKRun(KRunModule.java:80)
  while locating org.kframework.krun.api.KRun
  while locating org.kframework.krun.api.KRun annotated with @org.kframework.utils.inject.Main()
    for parameter 3 at org.kframework.krun.KRunFrontEnd.<init>(KRunFrontEnd.java:462)
  while locating org.kframework.krun.KRunFrontEnd
  while locating org.kframework.main.FrontEnd
  at org.kframework.utils.inject.CommonModule.tool(CommonModule.java:25)
  while locating org.kframework.main.Tool
    for field at org.kframework.main.Tool.instance(Tool.java:6)
Caused by: org.kframework.utils.errorsystem.KExceptionManager$KEMException
    at org.kframework.utils.errorsystem.KExceptionManager.registerInternal(KExceptionManager.java:86)
    at org.kframework.utils.errorsystem.KExceptionManager.register(KExceptionManager.java:75)
    at org.kframework.utils.errorsystem.KExceptionManager.register(KExceptionManager.java:66)
    at org.kframework.utils.errorsystem.KExceptionManager.registerCriticalError(KExceptionManager.java:43)
    at org.kframework.utils.options.DefinitionLoadingOptions.definition(DefinitionLoadingOptions.java:43)
    at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:104)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
    at com.google.inject.Scopes$1$1.get(Scopes.java:65)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
    at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
    at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
    at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
    at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection$1.call(InjectionRequestProcessor.java:116)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection$1.call(InjectionRequestProcessor.java:110)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection.injectMembers(InjectionRequestProcessor.java:110)
    at com.google.inject.internal.InjectionRequestProcessor.injectMembers(InjectionRequestProcessor.java:78)
    at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:170)
    at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
    at com.google.inject.Guice.createInjector(Guice.java:95)
    at com.google.inject.Guice.createInjector(Guice.java:72)
    at com.google.inject.Guice.createInjector(Guice.java:62)
    at org.kframework.main.Main.main(Main.java:83)

2) Error in custom provider, org.kframework.utils.errorsystem.KExceptionManager$KEMException
  at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
  at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
  while locating org.kframework.kil.loader.Context
  while locating org.kframework.kil.loader.Context annotated with @org.kframework.utils.inject.Main()
    for parameter 4 at org.kframework.krun.KRunFrontEnd.<init>(KRunFrontEnd.java:462)
  while locating org.kframework.krun.KRunFrontEnd
  while locating org.kframework.main.FrontEnd
  at org.kframework.utils.inject.CommonModule.tool(CommonModule.java:25)
  while locating org.kframework.main.Tool
    for field at org.kframework.main.Tool.instance(Tool.java:6)
Caused by: org.kframework.utils.errorsystem.KExceptionManager$KEMException
    at org.kframework.utils.errorsystem.KExceptionManager.registerInternal(KExceptionManager.java:86)
    at org.kframework.utils.errorsystem.KExceptionManager.register(KExceptionManager.java:75)
    at org.kframework.utils.errorsystem.KExceptionManager.register(KExceptionManager.java:66)
    at org.kframework.utils.errorsystem.KExceptionManager.registerCriticalError(KExceptionManager.java:43)
    at org.kframework.utils.options.DefinitionLoadingOptions.definition(DefinitionLoadingOptions.java:43)
    at org.kframework.utils.inject.DefinitionLoadingModule.context(DefinitionLoadingModule.java:29)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:601)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:104)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
    at com.google.inject.Scopes$1$1.get(Scopes.java:65)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
    at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
    at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:84)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
    at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
    at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
    at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
    at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:89)
    at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:98)
    at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
    at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:53)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection$1.call(InjectionRequestProcessor.java:116)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection$1.call(InjectionRequestProcessor.java:110)
    at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
    at com.google.inject.internal.InjectionRequestProcessor$StaticInjection.injectMembers(InjectionRequestProcessor.java:110)
    at com.google.inject.internal.InjectionRequestProcessor.injectMembers(InjectionRequestProcessor.java:78)
    at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:170)
    at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
    at com.google.inject.Guice.createInjector(Guice.java:95)
    at com.google.inject.Guice.createInjector(Guice.java:72)
    at com.google.inject.Guice.createInjector(Guice.java:62)
    at org.kframework.main.Main.main(Main.java:83)

2 errors
    at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:435)
    at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:175)
    at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
    at com.google.inject.Guice.createInjector(Guice.java:95)
    at com.google.inject.Guice.createInjector(Guice.java:72)
    at com.google.inject.Guice.createInjector(Guice.java:62)
    at org.kframework.main.Main.main(Main.java:83)
```

**Comment by dwightguth** [comment]
This should be fixed by #774. Please rebase against the latest unified
builtins and tell me if it recurs.

**Comment by yilongli** [comment]
I see. I was working on Andrei's branch and didn't see the fix.
